arch/Alpha/AlphaInstPrinter.c

```diff
@@ -0,0 +1,100 @@
+/* Capstone Disassembly Engine */
+/* By Dmitry Sibirtsev <sibirtsevdl@gmail.com>, 2023 */
+
+#ifdef CAPSTONE_HAS_ALPHA
+
+#include <platform.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "../../utils.h"
+#include "../../MCInstPrinter.h"
+
+#include "AlphaLinkage.h"
+#include "AlphaMapping.h"
+
+static const char *getRegisterName(unsigned RegNo);
+
+static void printInstruction(MCInst *, uint64_t, SStream *);
+
+static void printOperand(MCInst *MI, int OpNum, SStream *O);
+
+
+#define GET_INSTRINFO_ENUM
+
+#include "AlphaGenInstrInfo.inc"
+
+#define GET_REGINFO_ENUM
+
+#include "AlphaGenRegisterInfo.inc"
+
+static inline void fill_alpha_register(MCInst *MI, uint32_t reg)
+{
+	if (!(MI->csh->detail == CS_OPT_ON && MI->flat_insn->detail))
+		return;
+	cs_alpha *alpha = &MI->flat_insn->detail->alpha;
```

Comment:
Please checkout `Mapping.*`. We have a lot of helper functions for those checks and member access stuff (like `detail_is_set(MI)` or `<ARCH>_get_detail()`, `get_detail_op` etc.)